core/modules/filters/kin.js

```diff
@@ -0,0 +1,96 @@
+/*\
+title: $:/core/modules/filters/kin.js
+type: application/javascript
+module-type: filteroperator
+
+Finds out where a tiddler originates from and what other tiddlers originate from it
+
+\*/
+(function() {
+
+	/*jslint node: true, browser: true */
+	/*global $tw: true */
+	"use strict";
+
+	function collectTitlesRecursively(baseTiddler,baseTitle,options) {
+		var titlesPointingFromBase = [],
+			titlesPointingToBase = [];
+
+		function addToResultsIfNotFoundAlready(list,title) {
+			if(list.indexOf(title) !== -1) {
+				return false;
+			}
+			list.push(title);
+			return true
+		}
+
+		function collectTitlesPointingFrom(tiddler,title,currentDepth) {
+			if((options.depth) && (currentDepth++ > options.depth)) {
```

Comment:
As a matter of defensive programming style, I'd advise against using the increment operator in this way (the increment will only be performed if that part of the expression is evaluated, and JavaScripts expression shortcutting rules mean that that is not always obvious). It's clearer to perform the increment in a separate statement.